### PR TITLE
Celestia Contributing doc

### PR DESCRIPTION
**Pull request by MSevey**
This PR creates an org level `CONTRIBUTING.md` doc. 

If there is no `CONTRIBUTING.md` doc in a repository, this one will be used.  

This is an initial draft that focuses on some key parts of our Github processes. In the future this doc can be expanded to contain more information around our engineering standards.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (musalbas) [inline CONTRIBUTING.md:11]: s/organization/project/

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:59]: I'll update to `production` repos since I think I made that distinction somewhere else.

**Comment by MSevey** [comment]
A follow up will be to pull in pieces from the `CONTRIBUTING.md` file that is reference in the onboarding docs (currently the core repo's fork of tendermint `CONTRIBUTING.md`)

https://github.com/celestiaorg/celestia-core/blob/master-deprecated/CONTRIBUTING.md

File: CONTRIBUTING.md
Comment (SweeXordious) [inline CONTRIBUTING.md:6]:
```suggestion
In this guide, you will find information about how Celestia manages the
```

File: CONTRIBUTING.md
Comment (SweeXordious) [inline CONTRIBUTING.md:53]:
```suggestion
oriented reviews mean that there is a clear action step for the developer of
```

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:3]: [nit] thank you shouldn't be capitalized

```suggestion
Welcome and thank you for contributing to building Celestia, the first modular
```

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:7]: [nit] engineering standards doesn't need to be capitalized

```suggestion
`CelestiaOrg` Github as well as expectations around our engineering standards.
```

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:21]:
```suggestion
requests (i.e. adding labels). Teams are given write access to the repositories
```

File: CONTRIBUTING.md
Comment (SweeXordious) [inline CONTRIBUTING.md:37]: [Non Blocking][Question]
Should we specify which one is favorable? dev branches or forks? or leave it to the devs to decide?

File: CONTRIBUTING.md
Comment (SweeXordious) [inline CONTRIBUTING.md:55]: [Non Blocking][Suggestion]
@rootulp introduced a beautiful notation showing the intent of the feedback. 
For example: `[Non Blocking][Question]` shows that it's a non-blocking question.
I am not sure if this is formalized somewhere, rootul can correct me, but I guess if it is enforced organization wide, it would be very beneficial.

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:38]: [proposal] if we encourage pushing development branches directly to the celestiaorg repo, I think we should also encourage repo maintainers to enable this setting

Screen Shot 2022-10-17 at 11 29 05 AM


so that we can avoid instances of https://github.com/celestiaorg/celestia-core/issues/830 . Note that we may still end up with unmerged stale branches on the repo so repo maintainers may need to manually delete stale branches.

File: CONTRIBUTING.md
Comment (SweeXordious) [inline CONTRIBUTING.md:60]:
```suggestion
All repos have the following branch protections requirements:

- 2 approvals
```
[Non Blocking][Question]
Does it make sense to add the markdown linter CI for this repo too? 
https://github.com/celestiaorg/celestia-app/blob/main/.github/workflows/markdown-lint.yml

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:43]: [optional]

```suggestion
Celestia has a culture of prioritizing the review of PRs. This prioritization
focuses on unblocking others and finishing existing work before starting new
work. 
```

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:55]: [typo]

```suggestion
and statements should be avoided. Being clear when a comment is a blocking change,
```

**Review by SweeXordious — COMMENTED** [review]
Thanks a lot for helping define org wide contribution rules. 
Left some non-blocking nits and questions :cat:

File: CONTRIBUTING.md
Comment (rootulp) [reply CONTRIBUTING.md:9]: There exists a [Slack thread](https://celestia-team.slack.com/archives/C01JZFUPG48/p1662632401283949) on this and there was rough consensus on squash and merge.

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:19]: [optional] it seems strange for triage to be capitalized here. If we want to emphasize that this is an explicit Github permission, proposal to use bold

```suggestion
By default, all teams have **triage** access to all repositories in the
```

**Review by rootulp — APPROVED** [review]
Overall LGTM, thanks for proposing

File: CONTRIBUTING.md
Comment (adlerjohn) [inline CONTRIBUTING.md:11]: Technically the org name is explicitly and intentionally all lower-case b/c Go.

```suggestion
The `celestiaorg` Github uses teams to manage access to the organization and
```

File: CONTRIBUTING.md
Comment (adlerjohn) [inline CONTRIBUTING.md:42]: Celestia is a protocol, it doesn't have a culture. How can this be worded better?

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:38]: Yes I plan to have a list of repo setting in here as well and will include that one for sure.

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:19]: The reason it is capitalized is because it is capitalized in the permissions.

File: CONTRIBUTING.md
Comment (tzdybal) [inline CONTRIBUTING.md]: Are we actually going to use groups? Last time we failed and abandoned this idea.

File: CONTRIBUTING.md
Comment (tzdybal) [reply CONTRIBUTING.md:42]: maybe:
> Celestia Labs has a culture of prioritizing PRs.

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:37]: I think it is fine to be personal preference.

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md]: yes, this is on my todo list

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md]: @SweeXordious can you maybe start a thread in Slack to discuss? I'm a fan of having some sort of standard.  **REQ** **F/U** and **nit**  are other common ones. Shorter to write. Req is blocking and required, f/u is a recommended followup so not blocking, and nit is more personal preference and something to consider but not blocking and the dev can decide if they want to handle it or not.

File: CONTRIBUTING.md
Comment (rootulp) [reply CONTRIBUTING.md]: > Celestia has a culture of prioritizing PRs.

I think the existing first sentence is confusing because I don't know which one is higher priority: reviewing other people's PRs or working on my own PRs. My suggestion may have misinterpreted the order of priorities. I infer the priorities as:

1. Review all PRs that have requested my review
2. Continue working on PRs that I have proposed
3. Only if 1 and 2 are done, start new work

If that's the case, I don't think the existing section is clear and would prefer disambiguation on "prioritizing PRs".

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:70]: [non-blocking]

> These live in the code under `docs/adr`

I'm on-board with having ADRs in a consistent location across repos but flagging that this isn't the case today:
1. https://github.com/celestiaorg/celestia-app/tree/main/docs/architecture
2. https://github.com/celestiaorg/celestia-core/tree/v0.34.x-celestia/docs/celestia-architecture
3. https://github.com/celestiaorg/rollmint/tree/main/docs/lazy-adr

celestia-node conforms https://github.com/celestiaorg/celestia-node/tree/main/docs/adr but it's a bit difficult for celestia-core to conform because there are already Tendermint ADRs in https://github.com/celestiaorg/celestia-core/tree/v0.34.x-celestia/docs. If we're okay with an exception for celestia-core, (i.e. `docs/celestia-adr`) then I think we can update the non-conformant repos in follow-up PRs.

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:90]: [uber nit] I don't think "always" is correct here. For example: if a critical bug is discovered, repo maintainers may need to drop everything they're working on to resolve the bug.

```suggestion
start a new PR or work on closing out an existing PR, you should usually choose
```

File: CONTRIBUTING.md
Comment (rootulp) [inline CONTRIBUTING.md:99]: 👍

**Review by rootulp — APPROVED** [review]

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:70]: Yup thanks for this. As an fyi once this is merge I had a follow up to do a full organization audit to address discrepancies 👍

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:90]: Yea I typically like to document the happy case, otherwise you open the door to a bajillion edge cases :-) 

But I agree, I'm not going to focus on closing out a documentation PR if there is a critical bug haha

**Review by liamsi — COMMENTED** [review]
Awesome to see this finally happening 🙌 left some minor nits/comments. Generally, looks good to me :)

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:54]: ah yes, good point

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md:76]: They can be yes! Right now many of the repos have Issue templates, but we can pull the common ones into this repo so there is a nice consistent standard for things like ADRs at least.

File: CONTRIBUTING.md
Comment (MSevey) [reply CONTRIBUTING.md]: Will update once conversation is resolved in slack.  Resolving here for now.

**Review by rootulp — APPROVED** [review]
Thanks for addressing all my comments! Markdownlint fails [here](https://github.com/celestiaorg/.github/actions/runs/3282066100/jobs/5404947737#step:3:44) but this PR LGTM.

**Review by rootulp — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

**Pull request merged**
